cvmfs/repository.py

```diff
@@ -227,145 +107,64 @@ def verify(self, public_key_path):
             raise RepositoryVerificationFailed("Certificate doesn't fit", self)
         return True
 
-
-    def catalogs(self, root_catalog = None):
-        return CatalogTreeIterator(self, root_catalog)
-
-
     def has_repository_type(self):
         return hasattr(self, 'type') and self.type != 'unknown'
 
-
     def has_history(self):
         return self.manifest.has_history()
 
-
     def retrieve_history(self):
         if not self.has_history():
             raise HistoryNotFound(self)
         history_db = self.retrieve_object(self.manifest.history_database, 'H')
         return History(history_db)
 
-    def switch_revision(self, revision):
+    def get_current_revision(self):
+        return self._get_revision_by_number(self.manifest.revision)
+
+    def get_revision(self, revision_data):
+        if isinstance(revision_data, int):
```

Comment:
For my own surprise I've seen many APIs doing such a thing in python... so I guess is the way to go...